frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/PipelineRunDrawerRightTabs.tsx

```diff
@@ -35,47 +24,64 @@ const PipelineRunDrawerRightTabs: React.FC<PipelineRunDrawerRightTabsProps> = ({
   task,
   executions,
 }) => {
-  const [selection, setSelection] = React.useState(PipelineRunNodeTabs.INPUT_OUTPUT);
+  const hasNoInputsOutputs = !task.inputs && !task.outputs;
+  const [selection, setSelection] = React.useState<string>(
+    hasNoInputsOutputs ? PipelineRunNodeTab.DETAILS : PipelineRunNodeTab.INPUT_OUTPUT,
+  );
   const taskExecution = executions.find(
     (e) => e.getCustomPropertiesMap().get('task_name')?.getStringValue() === task.name,
   );
 
-  const tabContents: Record<PipelineRunNodeTabs, React.ReactNode> = {
-    [PipelineRunNodeTabs.INPUT_OUTPUT]: (
-      <SelectedNodeInputOutputTab task={task} execution={taskExecution?.toObject()} />
-    ),
-    // [PipelineRunNodeTabs.VISUALIZATIONS]: <>TBD 2</>,
-    [PipelineRunNodeTabs.DETAILS]: <SelectedNodeDetailsTab task={task} />,
-    [PipelineRunNodeTabs.VOLUMES]: <SelectedNodeVolumeMountsTab task={task} />,
-    [PipelineRunNodeTabs.LOGS]: <LogsTab task={task} />,
-    // [PipelineRunNodeTabs.POD]: <>TBD 6</>,
-    // [PipelineRunNodeTabs.EVENTS]: <>TBD 7</>,
-    // [PipelineRunNodeTabs.ML_METADATA]: <>TBD 8</>,
+  const tabs: Record<string, { title: string; isDisabled?: boolean; content: React.ReactNode }> = {
+    [PipelineRunNodeTab.INPUT_OUTPUT]: {
+      title: 'Input/Output',
+      isDisabled: hasNoInputsOutputs,
```

Comment:
do you need to put hasNoInputsOutputs as a variable, can it just be inline

Reply:
Its used in the condition for the default tab state as well, which is why I made a variable for it. I can repeat the condition twice if you want.

Reply:
no i just thought i didnt see it used twice, but im seeing it now so nvm